### PR TITLE
Feature/wmt 414/display arms assessments

### PR DESCRIPTION
**Pull request by andrew-js-wright**

### CODE REVIEW

File: app/services/workers/calculate-workload-points.js [hunk 1]
```diff
@@ -48,25 +46,46 @@ module.exports.execute = function (task) {
...
         var caseTypeWeightings = pointsConfiguration.values
-        var sdrConversionPoints = calculateSdrConversionPoints(workload.sdrConversionsLast30Days, caseTypeWeightings.pointsConfiguration.sdrConversion)
-        var sdrPoints = calculateSdrConversionPoints(workload.monthlySdrs, caseTypeWeightings.pointsConfiguration.sdr)
-        var paromsPoints = calculateParomPoints(workload.paromsCompletedLast30Days, caseTypeWeightings.pointsConfiguration.parom, caseTypeWeightings.pointsConfiguration.paromsEnabled)
-        var workloadPoints = calculateWorkloadPoints(workload, caseTypeWeightings)
+        var workloadPointsBreakdown = calculateWorkloadPoints(workload, caseTypeWeightings)
         return getAppReductionsPromise.then(function (reductions) {
           return getCmsAdjustmentPointsPromise.then(function (cmsAdjustments) {
-            var totalPoints = (workloadPoints + sdrConversionPoints + sdrPoints + paromsPoints + cmsAdjustments)
+            var totalPoints = workloadPointsBreakdown.total + cmsAdjustments
```
Comment (caoimheb) [inline app/services/workers/calculate-workload-points.js]: 👍

File: app/services/data/get-arms-totals.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const knex = require('knex')(knexConfig)
+
+module.exports = function (omKey, teamCode) {
+  var whereObject = {
+    'assessment_staff_key': omKey,
+    'assessmentent_team_key': teamCode // typo in extract
+  }
+
+  return knex.count('sentence_type as count').select('sentence_type')
+          .from(`${config.DB_STG_SCHEMA}.arms`)
```
Comment (ddebella) [inline app/services/data/get-arms-totals.js]: How would this work in Production?

File: test/helpers/data/staging-workload-helper.js [hunk 3]
```diff
@@ -50,28 +50,67 @@ const aliases = {
...
+  return stagingHelper.getTestCourtReport(omKey, undefined, teamCode)
 }
 
 module.exports.getTestInstitutionalReport = function (omKey = testOmKey) {
   return stagingHelper.getTestInstitutionalReport(omKey)
 }
 
+module.exports.getArmsData = function (omKey = testOmKey, teamCode = testTeamCode) {
+  var defaultArmsObject = {
+    assessmentDate: '09/10/2017',
```
Comment (ddebella) [inline test/helpers/data/staging-workload-helper.js]: Are we always defaulting to specific assessmentDate or is this to be changed?

File: app/services/data/get-arms-totals.js (hunk 2 above)
Comment (andrew-js-wright) [reply app/services/data/get-arms-totals.js]: Yea, in this case `STG` refers to the schema staging, which our ETL process loads data into, not the environment.

File: test/helpers/data/staging-workload-helper.js (hunk 3 above)
Comment (andrew-js-wright) [reply test/helpers/data/staging-workload-helper.js]: There is no date querying on the application side, this is just a test object to ensure we can correctly retrieve the property from the DB so the actual value will have no effect.

**Comment by andrew-js-wright** [comment]
Checks are failing until the probation rules library is published.

File: test/helpers/data/staging-workload-helper.js (hunk 3 above)
Comment (ddebella) [reply test/helpers/data/staging-workload-helper.js]: Yes, its the test helper 👍

File: app/services/data/get-arms-totals.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          .where(whereObject) // .andWhere(disposal_or_release_date is no more than 6 weeks ago)
```
Comment (caoimheb) [inline app/services/data/get-arms-totals.js]: Where is this `.andWhere` set?

File: app/services/data/get-arms-totals.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const knexConfig = require('../../../knexfile').app
+const knex = require('knex')(knexConfig)
+
+module.exports = function (omKey, teamCode) {
+  var whereObject = {
+    'assessment_staff_key': omKey,
+    'assessmentent_team_key': teamCode // typo in extract
+  }
+
+  return knex.count('sentence_type as count').select('sentence_type')
```
Comment (caoimheb) [inline app/services/data/get-arms-totals.js]: Minor: Capitalise keyword

File: app/services/data/get-arms-totals.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          .groupBy('sentence_type')
+  .then(function (results) {
+    var armsTotals = {
+      community: 0,
+      license: 0
+    }
+
+    results.forEach(function (row) {
+      switch (row.sentence_type) {
+        case 'Community':
```
Comment (caoimheb) [inline app/services/data/get-arms-totals.js]: We have generally been using all caps on these keywords. Are we sure this is the case they will be coming in in the extract?

File: app/services/data/get-arms-totals.js (hunk 4 above)
Comment (andrew-js-wright) [reply app/services/data/get-arms-totals.js]: This is an overhang, I've confirmed with Kristy that we'll only get ARMs assessments from the last 6 weeks so I'll remove this comment, cheers.

File: app/services/data/get-arms-totals.js (hunk 6 above)
Comment (andrew-js-wright) [reply app/services/data/get-arms-totals.js]: It was in the sample extract that Kristy sent to Tom. I'll move these to `const` so if they're different when the file lands we can easily change them.

**Pull request merged**